opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1000,21 +1001,32 @@ namespace Opm {
                 fip_.fip[i].resize(nc,0.0);
             }
 
-            for (int c = 0; c < nc; ++c) {
-                const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
+            ElementContext elemCtx(ebosSimulator_);
+            auto elemIt = elemCtx.gridView().template begin</*codim=*/0>();
+            const auto& elemEndIt = elemCtx.gridView().template end</*codim=*/0>();
+            for (; elemIt != elemEndIt; ++elemIt) {
+                const auto& elem = *elemIt;
+                if (elem.partitionType() != Dune::InteriorEntity)
+                    continue;
+
+                elemCtx.updatePrimaryStencil(elem);
```

Comment:
I agree that it is not very intuitive, so let me try to explain: the stencil is needed because without it, the intensive quantities cannot be calculated. that said, updating the central part of the stencil for the ECFV discretization is very cheap: it basically just stores the Dune entity for the element.

If one is lazy and does not care too much about performance, one can use
```
elemCtx.updateAll(elem);
```

but this updates all data (i.e. intensive and extensive quantities) for all DOFs of the stencil on all substeps. since only the intensive quantities for the most recent substep and only for the central element is needed here, it is done like it is...